packages/spacecat-shared-data-access/src/models/site-candidate.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { hasText, isValidUrl } from '@adobe/spacecat-shared-utils';
+
+import { Base } from './base.js';
+
+export const DEFAULT_UPDATED_BY = 'spacecat';
+
+export const SITE_CANDIDATE_STATUS = {
+  DISCOVERED: 'DISCOVERED', // site candidate is discovered
+  PENDING: 'PENDING', // site candidate notification sent and waiting for human input
+  IGNORED: 'IGNORED', // site candidate discarded: not to be added to star catalogue
+  APPROVED: 'APPROVED', // site candidate is added to star catalogue
+};
+
+/**
+ * Creates a new Site Candidate.
+ *
+ * @param {object} data - site candidate data
+ * @returns {Readonly<SiteCandidate>} new site candidate
+ */
+const SiteCandidate = (data = {}) => {
+  const self = Base(data);
+  delete self.id; // no id property used in SiteCandidate modal
+
+  self.getBaseURL = () => self.state.baseURL;
+  self.getStatus = () => self.state.status;
+  self.getUpdatedBy = () => self.state.updatedBy;
+
+  self.setStatus = (status, updatedBy = DEFAULT_UPDATED_BY) => {
```

Comment:
i think setStatus should not apply a default updatedBy as it is  explicitly called and should not result in an unintended default updated by which may be unintended.

Reply:
that's actually a good point